src/kernel/network.cpp

```diff
@@ -55,211 +56,181 @@ CryptoKernel::Network::Network(CryptoKernel::Log* log,
-                            throw Peer::NetworkError();
-                        }
-
-                        const auto banIt = banned.find(it->first);
-                        if(banIt != banned.end()) {
-                            if(banIt->second > static_cast<uint64_t>(std::time(nullptr))) {
-                                log->printf(LOG_LEVEL_WARN,
-                                            "Network(): Disconnecting " + it->first + " for being banned");
-                                throw Peer::NetworkError();
-                            }
-                        }
-
-                        it->second->info["height"] = info["tipHeight"].asUInt64();
-
-                        for(const Json::Value& peer : info["peers"]) {
-                            sf::IpAddress addr(peer.asString());
-                            if(addr != sf::IpAddress::None) {
-                                if(!peers->get(dbTx.get(), addr.toString()).isObject()) {
-                                    log->printf(LOG_LEVEL_INFO, "Network(): Discovered new peer: " + addr.toString());
-                                    Json::Value newSeed;
-                                    newSeed["lastseen"] = 0;
-                                    newSeed["height"] = 1;
-                                    newSeed["score"] = 0;
-                                    peers->put(dbTx.get(), addr.toString(), newSeed);
-                                }
-                            } else {
-                                changeScore(it->first, 10);
-                                throw Peer::NetworkError();
-                            }
-                        }
-                    } catch(const Json::Exception& e) {
-                        changeScore(it->first, 50);
-                        throw Peer::NetworkError();
-                    }
-
-                    const std::time_t result = std::time(nullptr);
-                    it->second->info["lastseen"] = static_cast<uint64_t>(result);
-                } catch(const Peer::NetworkError& e) {
-                    log->printf(LOG_LEVEL_WARN,
-                                "Network(): Failed to contact " + it->first + ", disconnecting it");
-                    removals.insert(it->first);
-                }
-            }
-
-            for(const auto& peer : removals) {
-                const auto it = connected.find(peer);
-                peers->put(dbTx.get(), peer, it->second->info);
-                if(it != connected.end()) {
-                    connected.erase(it);
-                }
-            }
-
-            for(const auto& peer : peerInfos) {
-                peers->put(dbTx.get(), peer.first, peer.second);
-            }
+void CryptoKernel::Network::makeOutgoingConnectionsWrapper() {
+	while(running) {
+		makeOutgoingConnections();
+		std::this_thread::sleep_for(std::chrono::milliseconds(20));
```

Comment:
Maybe this should be longer: `makeOutgoingConnections` tries a bunch of peers. It's unlikely the state will change enough to be worth trying again 20ms later.